mobius-core/src/main/java/com/spotify/mobius/Mobius.java

```diff
@@ -258,8 +258,12 @@ private Builder(
       LoggingInit<M, F> loggingInit = new LoggingInit<>(init, logger);
       LoggingUpdate<M, E, F> loggingUpdate = new LoggingUpdate<>(update, logger);
 
+      First<M, F> first = loggingInit.init(checkNotNull(startModel));
```

Comment:
Not really a comment on this line but rather the file. Given this change, should we add new function 
`startWith(M model, Set<F> effects)`?

Reply:
What would be the point? Adding custom effects for a given startup of the loop? What is the 'semantic meaning' of that? Generally, I've been thinking that the initial effects are defined by the `Init` function, but this would then give another way of defining initial effects - is that a mostly good thing? It would seem to add flexibility at the cost of reducing understandability.

Reply:
Another way if so would be to remove the concept of `Init` entirely, and just allow passing in a model + effects to execute initially. That would lose some of the power of 'resume from point X', I think.

Reply:
Adding that method was actually the end goal, but I was planning on doing it in the next PR. Specifically the goal with this chain of changes was to move Init to MobiusLoop.Controller:

To start a loop manually (without a controller) you need to pass in a start model. If you want to pass in effects then you can't do that directly right now, you have to pass in an init to do that, which in a sense could be seen as the init being pre-composed with the startFrom. If the init was optional, then you wouldn't have to use that extra function to express start effects, but could easily add it if you want to (and in fact you could make a more powerful init that goes from any T to (M, Set<F>) instead of necessarily looking like init does now).

There is however as you say a very good case for init in "resume from point X" cases, but those really only exist in the controller where you need to start more than once and therefore need to bundle in a "resume from this point" strategy.

Reply:
I don't think `Init` should be optional - usually, if you have two ways of doing the same thing, that's a smell. There seems to be a development of a mental model where there is a clear set of layers - the store as the innermost one, then the loop, then the controller. I think if those layers are clearly defined in terms of responsibilities and value added, then perhaps it would be possible to say that `Init` belong in the controller, and the only way to start a loop is through a `startFrom(M, Set<F>)`. Have you done the exercise of thinking through and defining the purposes of these layers already?

Reply:
sorry yes, that is what i meant, that the only way to start a loop should be through `startFrom(M, Set<F>)` - i want to remove init from the loop itself completely (except possibly leaving a deprecated method behind until we do a major version bump).

what i meant was that if you want to compose loops together then you have a clear way to do any kind of initialization you want without having to do a dance with `init(M)` just to pass in effects.